### PR TITLE
gene set summary progress

### LINKED ISSUE #2057: Backend support for mean expression of a gene set
1. Mean expression is the default once a user expands a gene set, we want coloring by the mean of a gene set to be fast so ok to add time to initial load (ok for it to take a short amount load time once a user switches from looking at the mean to looking at inidividual genes; however we expect this to be a common pattern of switching between mean and individual genes)
2. We should cache a small number of gene sets (~20) so users can analyze some gene sets at interactive speeds
3. System should not blow up if there are a large number of gene sets (like 1,000. we discussed it might be possible to load in the background)

@bkmartinjr let me know if I missed anything here or if you have any additional thoughts or questions. Also added to requirements doc.

### LINKED ISSUE #2100: front-end reducer / caching / crossfilter integration for geneset summary
<eom>

### PR DESCRIPTION
**Pull request by bkmartinjr**
Several changes:
* front-end work on annoMatrix and other plumbing to enable use of gene set summaries in the UI
* backend routes revised - better / simpler API, and better caching.
* unit tests for both
* lint fixes to comply with latest eslint checks

Fixes #2057 
Fixes #2100

### CODE REVIEW

File: client/src/components/leftSidebar/topLeftLogoAndTitle.js [hunk 1]
```diff
@@ -13,15 +13,14 @@ const DATASET_TITLE_FONT_SIZE = 14;
 @connect((state) => {
   const { corpora_props: corporaProps } = state.config;
   const correctVersion =
-    ["1.0.0", "1.1.0"].indexOf(
-      corporaProps?.version?.["corpora_schema_version"]
-    ) > -1;
+    ["1.0.0", "1.1.0"].indexOf(corporaProps?.version?.corpora_schema_version) >
```
Comment (colinmegill) [inline client/src/components/leftSidebar/topLeftLogoAndTitle.js:16]: Can this be removed or filed to be removed?

**Review by colinmegill — APPROVED** [review]
LGTM

File: client/src/components/leftSidebar/topLeftLogoAndTitle.js (hunk 1 above)
Comment (bkmartinjr) [reply client/src/components/leftSidebar/topLeftLogoAndTitle.js:16]: See #2038

**Review by MDunitz — APPROVED** [review]
lgtm

**Pull request merged**